src/NuGet.Core/NuGet.Logging/Logger.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Logging
+{
+    public class Logger
+    {
+        private static ILogger _logger = new NullLogger();
+
+        public static ILogger Instance
+        {
+            get
+            {
+                return _logger;
+            }
+            set
+            {
+                _logger = value ?? new NullLogger();
```

Comment:
Right, logger instance is initialized with NullLogger by default to avoid nullrefs. I'm fine with that.
But "set" behavior in Logger.Instance = null is odd. I suggest throwing ArgumentNullException if someone intentionally attempts to nullify the Instance. Thus we avoid some tricky bugs in future.